frontend/src/ui/section/event-detail/EventDetail.tsx

```diff
@@ -1,62 +1,109 @@
+import DOMPurify from 'dompurify';
+import { EmojiConvertor } from 'emoji-js';
+
+import { getEventDetails } from '@/modules/events/application/get-details/getEventDetails';
+import { DetailedEvent } from '@/modules/events/domain/Event';
+import { createEventRepository } from '@/modules/events/infrastructure/EventRepository';
 
 import '@/ui/section/event-detail/EventDetail.scss';
 
 export function EventDetail() {
   const { id } = useParams();
-  const {
-    title,
-    month,
-    day,
-    weekday,
-    location,
-    description,
-    time,
-    petsAllowed,
-  } = eventMock;
-
-  // TODO: When backend is ready check if event exist or not and modify this condition
+  const [eventDetails, setEventDetails] = useState<DetailedEvent>();
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const eventRepository = createEventRepository();
+
+  const emojiConvertor = new EmojiConvertor.EmojiConvertor();
+  emojiConvertor.replace_mode = 'unified';
+
+  const load = async (eventId?: string) => {
+    if (eventId) {
+      setHasError(false);
+      try {
+        const event: DetailedEvent = await getEventDetails(
+          eventRepository,
+          eventId,
+        );
+        setEventDetails(event);
+      } catch (error) {
+        setHasError(true);
+      }
+      setIsLoading(false);
+    }
+  };
+
+  function sanitizeHtml(text: string) {
+    return (
+      DOMPurify.sanitize(text),
+      {
+        ALLOWED_TAGS: ['br', 'p'],
+      }
+    );
+  }
+
+  useEffect(() => {
+    eventDetails?.description ? sanitizeHtml(eventDetails?.description) : null;
+  }, [eventDetails?.description]);
+
+  useEffect(() => {
+    load(id);
+  }, [id]);
+
   return (
-    <div className="eventDetail">
-      {id ? (
-        <EventInfo
-          title={title}
-          month={month}
-          day={day}
-          isBoldTitle
-          weekday={weekday}
-          colour="#000000"
-        />
+    <Section
+      className="eventDetail"
+      refresh={() => load(id)}
+      isLoading={isLoading}
+      shouldRefresh={hasError}
+    >
+      {eventDetails ? (
+        <>
+          <EventInfo
+            title={eventDetails?.summary?.name}
+            month={eventDetails?.summary?.month}
+            day={eventDetails?.summary?.day}
+            isBoldTitle
+            weekday={eventDetails?.summary?.weekday}
+            colour={eventDetails?.summary?.colour}
+          />
+          <div className="eventDescription">
+            <div
+              className="eventDescription__description"
+              dangerouslySetInnerHTML={{
+                __html: emojiConvertor.replace_colons(
+                  eventDetails?.description,
+                ),
+              }}
+            ></div>
+          </div>
+          <div>
+            <IconWithText
+              icon={<img alt="Calendar" src={calendarClockImage} />}
+            >
+              {eventDetails.time}
```

Comment:
Be solid, all access to the eventDetails properties are with ?. like   title={eventDetails?.summary?.name}
            month={eventDetails?.summary?.month}
            day={eventDetails?.summary?.day}
            
            but for the rest you dont use Nullish coalescing operator.
            
           Apply the same logic without  Nullish coalescing operator for all properties

Reply:
done!